stataggregator/countaggregatorconfig.go

```diff
@@ -0,0 +1,54 @@
+package stataggregator
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	stat "github.com/asecurityteam/component-stat"
+	"github.com/rs/xstats"
+)
+
+// CountAggregatorConfig is for configuration a CountAggregator.
+type CountAggregatorConfig struct {
+	FlushInterval time.Duration `description:"Frequencing of aggregating metrics before sending."`
+	StatConfig    *stat.DatadogConfig
+}
+
+// Name of the configuration as it might appear in config files.
+func (*CountAggregatorConfig) Name() string {
+	return "statcountaggregator"
+}
+
+// CountAggregatorComponent implements the settings.Component interface for
+// a countaggregator.
+type CountAggregatorComponent struct {
+	StatComponent *stat.DatadogComponent
+}
+
+// Settings generates a config with default values applied.
+func (c *CountAggregatorComponent) Settings() *CountAggregatorConfig {
+	return &CountAggregatorConfig{
+		FlushInterval: 10 * time.Second,
+		StatConfig:    c.StatComponent.Settings(),
+	}
+}
+
+// New creates a configured countaggregator that wraps around a stats client.
+func (c *CountAggregatorComponent) New(ctx context.Context, conf *CountAggregatorConfig) (xstats.XStater, error) {
+
+	stater, err := c.StatComponent.New(ctx, conf.StatConfig)
+	if err != nil {
+		return nil, fmt.Errorf("bad stat config")
```

Comment:
What errors might `StatComponent.New` return? a "bad stat config" might happen in StatComponent.New, so if you wan to distinguish your error message, then you might want to add context about the count aggregator, otherwise return the error that StatComponent.New returns

Reply:
Agreed. I would prefer the original error be returned here. You can decorate it with your own error if you want, but I see no reason to do so.

Reply:
yea idk what i was thinking.... lol